api/views/diagnostic.py

```diff
@@ -346,3 +358,22 @@ def _update_location_data(canteens, locations_csv_str):
                     canteen.save()
         except Exception as e:
             logger.error(f"Error while updating location data : {repr(e)} - {e}")
+
+    @staticmethod
+    def _get_manager_emails(row):
+        manager_emails = row.split(",")
+        normalized_emails = list(map(lambda x: get_user_model().objects.normalize_email(x.strip()), manager_emails))
+        for email in normalized_emails:
+            validate_email(email)
+        return normalized_emails
+
+    @staticmethod
+    def _add_managers_to_canteen(emails, canteen):
+        for email in emails:
+            try:
+                AddManagerView.add_manager_to_canteen(email, canteen)
+            except IntegrityError as e:
```

Comment:
Est-ce qu'on veut catcher (that's definitely a french word...) toutes autres exceptions et les logger pour éviter les réponses 500 ?

Reply:
Normalement le catch dans la ligne 139 le prend non ?

```python
def _treat_csv_file(self, file):
    ...
    try:
        ...
        canteen = self._create_canteen_with_diagnostic(row, siret) # _add_managers_to_canteen appellé ici
        ...
    except Exception as e:
        # Gestion des erreurs générique
```